### DISCUSSION TITLE
How to turn `side-by-side` view off for a single command?

### DISCUSSION THREAD

**Discussion by jakeboone02** [dandavison/delta#1522] [Q&A]
Is there a way I can pipe raw `git diff` output to `delta` and turn side-by-side view _off_ if `side-by-side = true` is configured in my `.gitconfig`? This is the full config file:

```gitconfig
[core]
	pager = delta
[interactive]
	diffFilter = delta --color-only
[delta]
	navigate = true
	side-by-side = true
[merge]
	conflictstyle = diff3
[diff]
	colorMoved = default
```

I thought I could run something like this:

```shell
git diff HEAD~1..HEAD | delta --side-by-side=false
# or
cat somefile.diff | delta --side-by-side=false
```

...but the `--side-by-side` flag doesn't accept values:

```
error: The value 'false' was provided to '--side-by-side' but it wasn't expecting any more values
```

EDIT:

I just saw [the side-by-side page on the website](https://dandavison.github.io/delta/side-by-side-view.html), but this "deactivate" command didn't seem to have any effect:

```shell
export DELTA_FEATURES=+side-by-side # activate
export DELTA_FEATURES=+             # deactivate
```

Also tried these with no luck:

```shell
DELTA_FEATURES='' git diff HEAD~1..HEAD
git diff HEAD~1..HEAD | DELTA_FEATURES='' delta
```

**Comment by weirdgiraffe** [comment]
It works as expected, if you have this feature turned off by default, i.e. in your git config you should have:

```
core]
	pager = delta
[interactive]
	diffFilter = delta --color-only
[delta]
	navigate = true
[merge]
	conflictstyle = diff3
[diff]
	colorMoved = default
```

Then you can go and do diff in a single view mode:


```
git diff HEAD~1..HEAD
```

And if you want side by side, then just do

```
DELTA_FEATURES=+side-by-side git diff HEAD~1..HEAD
```

**Reply by jakeboone02** [reply]
Right, but I'm looking for a way to turn it off, overriding a `.gitconfig` that has it turned on.

I don't want it off by default, just every now and then.

**Reply by dandavison** [reply]
Hm, yes I've documented my current solution at https://dandavison.github.io/delta/tips-and-tricks/toggling-delta-features.html but, yes, it currently requires `side-by-side` to not be set in git config.

Definitely room for improvement here, regarding the priority / clashes, and perhaps someone could add a command to delta itself that spits out appropriate shell code to be eval'd.

**Reply by weirdgiraffe** [reply]
I just use aliases `gdiff` for normal diff and `gsdiff` for side by side diff.

**Comment by th1000s** [comment]
Related to `--no-side-by-side` and similar "enable-only-flags", there is a long-standing open issue in the clap command line parser delta uses: [Add support for automatic negation flags: clap-rs/clap#815](https://github.com/clap-rs/clap/issues/815).

**Reply by weirdgiraffe** [reply]
I personally have an opinion that flags should be off by default. Because it's intuitive and consistent. So I'm not a big fan of adding --no-whatever flags. What I would like to is to be able to specify an alias in git config for side by side diff and simple diff. But it seems that current way of integration doesn't allow it

**Reply by dandavison** [reply]
Like this?

```gitconfig
[alias]
    diff-side-by-side = -c delta.features=side-by-side diff
```

**Reply by weirdgiraffe** [reply]
Oh! Thank you very very much

**Comment by almereyda** [comment]
Looking forward to any workaround that might come up which allows to temporarily disable side by side diffs for a delta configuration that has them set up by default.
